### PR TITLE
[TestCase] Fix test cases for `azurerm_virtual_network_gateway`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description
As the test cases of `azurerm_virtual_network_gateway` are failed in Teamcity for a long time, so I submitted this PR to fix them. This PR is majorly to fix the test cases which start with "TestAccVirtualNetworkGateway_" except "TestAccVirtualNetworkGateway_edgeZone". So please skip "TestAccVirtualNetworkGateway_edgeZone" since it also failed with same error in Teamcity Daily Run and this test case needs to be fixed with more information from service team.
image
image

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_virtual_network_gateway` - fix acctests

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: .teamcity/components/settings.kt [hunk 1]
```diff
@@ -144,7 +144,7 @@ var serviceTestConfigurationOverrides = mapOf(
         "networkfunction" to testConfiguration(locationOverride = LocationConfiguration("westus2", "eastus2", "westeurope", false)),
 
         // Network Regional Tire Public IP is only available in
-        "network" to testConfiguration(locationOverride = LocationConfiguration("westeurope", "eastus2", "westus", false)),
+        "network" to testConfiguration(locationOverride = LocationConfiguration("eastus", "westus", "westeurope", false)),
```
Comment (jackofallops) [inline .teamcity/components/settings.kt:147]: Why the need to change the region? We keep to `westeurope` as the Primary unless there's a specific reason not to.

**Review by jackofallops — COMMENTED** [review]
Hi @neil-yechenwei - just one minor question, otherwise this looks fine.

**Comment by neil-yechenwei** [comment]
@jackofallops , thanks for the comments. I updated PR. Please take another look. The test case in below latest test result I just now re-triggered is also failed with same error in Teamcity Daily Run. So it's not related with this PR.

image

File: .teamcity/components/settings.kt (hunk 1 above)
Comment (neil-yechenwei) [reply .teamcity/components/settings.kt:147]: Seems it was unstable in this region. OK. I will change it back.

**Comment by neil-yechenwei** [comment]
@jackofallops , thanks for the comments. Please take another look.

**Review by stephybun — APPROVED** [review]
Thanks @neil-yechenwei LGTM 👍

**Pull request merged**